### PR TITLE
Cursor

### PR DESCRIPTION
**Pull request by harishkrupo**
Changes in this commit:
- Add helper functions to maintain the list of created
  layers in compositor-iahwc.c
- Add support for setting raw pixel data in the linux
  frontend
- Add support for mmapping buffers.
- Map the raw pixel data and copy it to a gbm bo.
- Delete the weston patches maintained in this repo as
  this tree now contains them:
  https://github.com/intel/external-weston/tree/iahwc-plugin
- Readme.md change to reflect the above change
- modify travis.yml to build without the build_script.sh

Signed-off-by: Kalyan Kondapally <kalyan.kondapally@intel.com>
Signed-off-by: Harish Krupo <harish.krupo.kps@intel.com>

### CODE REVIEW

**Comment by harishkrupo** [comment]
@kalyankondapally I have squashed some of the stray commits into a single one. 
Is this Okay?

**Comment by harishkrupo** [comment]
Still need to delete the patches.

**Comment by csravelar** [comment]
Can we add a description for this? Would help when trying to review

**Comment by harishkrupo** [comment]
.Yeah, sure :) still making some amends.

**Comment by harishkrupo** [comment]
Okay, updated. Please let me know if this is missing some information.

File: wsi/pixelbuffer.h [hunk 1]
```diff
@@ -59,6 +59,8 @@ class PixelBuffer {
 
  private:
   bool needs_texture_upload_ = true;
+  uint32_t orig_width_, orig_height_, orig_stride_;
+
```
Comment (kalyankondapally) [inline wsi/pixelbuffer.h]: extra line

**Comment by kalyankondapally** [comment]
I didnt see the need to disable cache, so removing that one.

**Pull request closed without merging**